src/alignment.js

```diff
@@ -82,7 +82,7 @@ var alignWithDOM = function(nodeName, key, statics) {
   if (currentNode && matches(currentNode, nodeName, key)) {
     matchingNode = currentNode;
   } else {
-    var existingNode = key && getChild(parent, key);
+    var existingNode = /** @type {Element} */(key && getChild(parent, key));
```

Comment:
That case is covered by getChild method itself https://github.com/google/incremental-dom/pull/112/files#diff-db8f78209fc0a7d78756881ed982a3c3R128

> On 20 Aug 2015, at 23:36, Justin Ridgewell <notifications@github.com> wrote:
> 
> In src/alignment.js <https://github.com/google/incremental-dom/pull/112#discussion_r37589166>:
> 
> > @@ -82,7 +82,7 @@ var alignWithDOM = function(nodeName, key, statics) {
> >    if (currentNode && matches(currentNode, nodeName, key)) {
> >      matchingNode = currentNode;
> >    } else {
> > -    var existingNode = key && getChild(parent, key);
> > +    var existingNode = /** @type {Element} */(key && getChild(parent, key));
> It's definitely necessary as getChild will lazily create a keyMap.
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/google/incremental-dom/pull/112/files#r37589166>.
>